qemu/tests/transfer_file_over_ipv6.py

```diff
@@ -73,7 +73,8 @@ def get_file_md5sum(file_name, session, timeout):
             params.get("os_type"),
             ip_version="ipv6",
             linklocal=link_local_ipv6_addr)
-
+        if link_local_ipv6_addr is False and addresses[vm] is None:
```

Comment:
Should this be `and` or `or`? I suppose each of them could fail the case.

Reply:
`link_local_ipv6_addr is False` means we will use remote ip address,in this case we will have the problem of not being able to get an IP address(`addresses[vm] is None`).So we need to use `and` here.

Reply:
@leidwang So if `link_local_ipv6_addr is False` then `addresses[vm]` is 100% `None`, right? then `and addresses[vm] is None` can be removed? One more unnecessary condition could reduce the efficiency of the code.

Reply:
@leidwang I have just went through the script, so there is a chance when `link_local_ipv6_addr is False` and you can successfully get a  `addresses[vm]`, and that is what you expected for a proper test environment, otherwise the case should be cancelled, is that the case? If yes then current implementation makes sense. But could we provide more information in the cancel message, like how should the tester found a proper environment or why current one can not fit the test requirement?

Reply:
Yes, you are right. The environment depends on which route your host belongs to, we can contact IT if we want our host to support remote IP addresses.